app/models/thread.php

```diff
@@ -0,0 +1,115 @@
+<?php 
+class Thread extends AppModel
+{
+    //Minimum Length Values
+    const MIN_TITLE_LENGTH = 1;
+
+    //Maximum Length Values
+    const MAX_TITLE_LENGTH = 30;
+
+    public $validation = array(
+        'title' => array(
+            'length' => array(
+                'validate_between', self::MIN_TITLE_LENGTH, self::MAX_TITLE_LENGTH),
+            )
+        );
+
+    public function create(Comment $comment)
+    {
+        $this->validate();
+        $comment->validate();
+
+        if ($this->hasError() || $comment->hasError()) {
+            throw new ValidationException('Invalid thread or comment');
+        }
+
+        $db = DB::conn();
+        
+        $date_created = date("Y-m-d H:i:s");
+
+        $params = array(
+        'title' => $this->title,
+        'created'=> $date_created
+        );
+
+        try {
+            $db->begin();
+            $db->insert('thread', $params);
```

Comment:
No need to use $params variable here po, you could directly put the array here :bow: